com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/handlers/v2/DescriptorHandler.java

```diff
@@ -20,43 +21,41 @@
 public class DescriptorHandler {
 
     public RequiredDependency findRequiredDependency(DeploymentDescriptor descriptor, String moduleName, String dependencyName) {
-        for (Module module : descriptor.getModules()) {
-            if (module.getName()
-                .equals(moduleName)) {
-                return findRequiredDependency(module, dependencyName);
-            }
-        }
-        return null;
+        return descriptor.getModules()
+            .stream()
+            .filter(module -> module.getName()
+                .equals(moduleName))
+            .findFirst()
+            .map(module -> findRequiredDependency(module, dependencyName))

```

Comment:
Why map after findFirst?

Reply:
In this specific example it wouldn't make a difference and both ways would work the same way. I just prefered mapping the Optional instead of the stream.

Reply:
Actually the latter is optimised (no need for N mappings when you can have only 1)